sklearn/preprocessing/imputation.py

```diff
@@ -299,13 +311,74 @@ def _dense_fit(self, X, strategy, missing_values, axis):
 
             return most_frequent
 
+    def _sparse_transform(self, X, valid_stats, valid_idx):
+        """transformer on sparse data."""
+        mask = _get_mask(X.data, self.missing_values)
+        indexes = np.repeat(np.arange(len(X.indptr) - 1, dtype=np.int),
+                            np.diff(X.indptr))[mask]
+
+        X.data[mask] = astype(valid_stats[indexes], X.dtype,
+                              copy=False)
+
+        mask_matrix = X.__class__((mask, X.indices.copy(),
+                                  X.indptr.copy()), shape=X.shape,
+                                  dtype=X.dtype)
+        mask_matrix.eliminate_zeros()  # removes explicit False entries
+        features_with_missing_values = mask_matrix.sum(axis=0).nonzero()[1]
```

Comment:
Oh. It's this line broken again. Apparently `.nonzero` returns a sequence of matrices. I don't get why that's useful, but anyway... :/

Stick `.A` before `.nonzero` and it should magically work.